debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotChangeEventSource.java

```diff
@@ -547,7 +548,9 @@ private void checkAndProcessStopFlag(P partition, OffsetContext offsetContext) {
                 nextDataCollection(partition, offsetContext);
             }
         }
-        notificationService.incrementalSnapshotNotificationService().notifyAborted(context, partition, offsetContext, stopped);
+        else {
+            notificationService.incrementalSnapshotNotificationService().notifyAborted(context, partition, offsetContext, stopped);
+        }
```

Comment:
Would it make sense to add a `continue;` statement in the right places and keep the previous code?